src/screens/Dashboard/Collection/index.js

```diff
@@ -61,37 +70,58 @@ const Collection = () => {
         setFile(null);
     };
 
-    const save = async () => {
-        if (imageUrl && title) {
-            let data = {};
-            let result = false;
-            data.title = title;
-            data.img = imageUrl;
-            // data.tokenAddress = token;
-            // data.marketplaceAddress = marketplace;
-            if (collectionId) {
-                data.collectionId = collectionId;
-                result = await Actions.updateCollection(data);
-            } else {
-                result = await Actions.saveCollection(data);
-            }
-            if (result) {
-                dispatch(Actions.getCollections());
-                if (collectionId) {
-                    toast.success("Updated successfully!", toastOptions);
-                } else {
-                    toast.success("Added successfully!", toastOptions);
-                }
-                setVisibleModal(false);
-            } else {
-                toast.error("Something went wrong!", toastOptions);
-                setVisibleModal(false);
-            }
-        } else {
-            toast.error("Fill in all fields completely!", toastOptions);
+    const save = () => {
+
+        if (!imageUrl || !title || !symbol || !initBaseURI || !maxSupply || !mintPrice) {
+            toast.error("Please input all fields!", toastOptions);
+            return;
         }
+
+        setVisibleModal(false);
+
+        const web3 = new Web3(Web3.givenProvider);
+        web3.eth.transactionBlockTimeout = 1000;
+
+        const _Contract = new web3.eth.Contract(
+            tokenManagerContractABI.abi,
+            process.env.REACT_APP_TOKENMANAGER_CONTRACT_ADDRESS
+        );
+
+        let toWei = web3.utils.toWei(mintPrice, 'ether');
+        _Contract.methods.deployCollection(title, symbol, initBaseURI, imageUrl, maxSupply, toWei)
+                                .send({ from: auth.address});
+        
+        _Contract.events.CollectionAdded().on('data', (event) => {
+            console.log(`collection deployed: ${event.returnValues._addr}`);
+            saveCollection(event.returnValues._addr);
+        })
```

Comment:
Should be in backend. Frontend doesn't listen smart contract events.
I have already explained. Backend does this instead.